### PR TITLE
Added tests for co.cask.cdap.api.common.Bytes

### PR DESCRIPTION
**Pull request by freddytuxworth**
### What is this PR for?
I analysed this codebase and found that the `Bytes` class could do with some more unit tests. I have written a few tests for this class with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/datasheet). Hopefully these tests will help you to detect any regressions caused by future code changes.

Please let me know if there are any other areas of this project which you'd like tests written for - I'd be more than happy to help.

### CODE REVIEW

File: cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java [hunk 1]
```diff
@@ -26,6 +31,583 @@
...
+    @Test
+    public void testAdd(){
+        final byte[] a = {1, 2, 3, 4};
+        final byte[] b = {5, 6, 7, 8};
+        final byte[] empty = {};
+
+        Assert.assertArrayEquals(new byte[] {1,2,3,4,5,6,7,8}, Bytes.add(a, b));
+        Assert.assertArrayEquals(a, Bytes.add(a, empty));
+        Assert.assertArrayEquals(empty, Bytes.add(empty, empty));
+
```
Comment (paulkennethkent) [inline cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java]: random new line

File: cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java [hunk 2]
```diff
@@ -26,6 +31,583 @@
...
+        Assert.assertArrayEquals(new byte[] {0, 0, 0, 0, 0, 0, 0, 0}, bytes);
+
+        bytes = new byte[8];
+        Assert.assertEquals(8, Bytes.putDouble(bytes, 0, 0.55d));
+        Assert.assertArrayEquals(new byte[] {63, -31, -103, -103, -103, -103, -103, -102}, bytes);
+
+        bytes = new byte[10];
+        Assert.assertEquals(9, Bytes.putDouble(bytes, 1, 0.55d));
+        Assert.assertArrayEquals(new byte[] {0, 63, -31, -103, -103, -103, -103, -103, -102, 0}, bytes);
+
```
Comment (paulkennethkent) [inline cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java]: random new line

File: cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java [hunk 3]
```diff
@@ -26,6 +31,583 @@
...
+        thrown.expect(NegativeArraySizeException.class);
+        Bytes.padHead(a, -10);
+    }
+
+    @Test
+    public void testPutBigDecimal() {
+        final byte[] bytes = {};
+        Bytes.putBigDecimal(bytes, 0, new BigDecimal("123456789.123456789"));
+
```
Comment (petersear) [inline cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java]: remove blank line

Comment (petersear) [reply cdap-api/src/test/java/co/cask/cdap/api/common/BytesTest.java]: There's no assert here or should it throw an exception?

**Review by petersear — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Pull request closed without merging**